src/foam/core/Validation.js

```diff
@@ -327,10 +327,36 @@ foam.CLASS({
               },
               errorString: `${this.label} is required`
             }
-          )
+          );
         }
         return ret;
       }
     }
   ]
 });
+
+foam.CLASS({
+  package: 'foam.core',
+  name: 'ValidationPasswordRefinement',
+  refines: 'foam.core.Password',
+
+  properties: [
+    {
+      name: 'minLength',
+      value: 6
+    },
+    {
+      class: 'FObjectArray',
+      of: 'foam.core.ValidationPredicate',
+      name: 'validationPredicates',
+      factory: function() {
+        if ( this.required ) {
+          var superFactory = foam.core.String.VALIDATION_PREDICATES.FACTORY;
+          return superFactory();
+        }
+        return [];
+      }
```

Comment:
Above compiles, but I am yet to test if return superFactory() does what I think.

Reply:
"compiles" is not much of a guarantee in JS.  If you haven't tested it yet then it's not ready to be submitted.  It's fine to start PRs before testing if you're looking for feedback on your implementation strategy, but be sure to mark such PRs as "DO NOT MERGE."

What are you trying to achieve?  It looks like you want the same validation predicates as foam.core.String but only if ths field is marked as required?  Is there any reason not to respect "required" in the foam.core.String validation predicates?

At the very least you probably have to do `foam.core.String.VALIDATION_PREDICATES.factory` and `superFactory.call(this);`  but I think we should re-think why we want to do this in the first place.

Reply:
First Objective: Simplify password validation to 6 char.
https://nanopay.atlassian.net/browse/CPF-2259 will correct the requirements. Which will be 7 char. But don't want to break anyone in production with a 6 char password right now.

So I added this here to confirm field validation on Passwords.

IF we do nothing but set the default minLength to 6, this breaks anywhere we are incorrectly using a User obj. Ex) CABankPADForm.js
validateInputs()  checks the errors on the user(this.viewData.user.errors_) despite only setting a first and last name.

Thus I tried to make my changes as unobtrusive as possible. Despite User having a password field and password is getting default setting is to have a minLength, I added that we only need this if field is required.

String (which Password extends) does not need to have this. If a minLength is set then there should be validation.

Reply:
AND

Sorry about not adding DNM to pr name.

Tested and changed. YOu were right:
foam.core.String.VALIDATION_PREDICATES.factory and superFactory.call(this);

Reply:
I'm not sure I follow why this is needed either. You should be able to simply set the minLength to 6 here and shouldn't need this additional code.

Also, I'm not sure we've really defined what the `required` flag should do wrt validation. Does it mean the property is set (i.e. enforces that hasOwnProperty returns true)? A property can be set to an empty string so what happens in that case? We should answer those before making use of it here.